gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -191,30 +191,44 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 	// Parse and run the files, construct *PV.
 	m2 := gno.NewMachineWithOptions(
 		gno.MachineOptions{
-			PkgPath:   "",
-			Output:    os.Stdout, // XXX
-			Store:     store,
-			Alloc:     store.GetAllocator(),
-			Context:   msgCtx,
-			MaxCycles: vm.maxCycles,
+			PkgPath:    "",
+			Output:     os.Stdout, // XXX
+			Store:      gnostore,
+			Alloc:      gnostore.GetAllocator(),
+			Context:    msgCtx,
+			MaxCycles:  vm.maxCycles,
+			GasMeter: ctx.GasMeter(),
 		})
-	defer m2.Release()
-	m2.RunMemPackage(memPkg, true)
+	defer func() {
+		if r := recover(); r != nil {
+			switch r.(type) {
+			case store.OutOfGasException: // panic in consumeGas()
+			  m2.Release()
+				panic(r)
```

Comment:
formatting is wrong, please run go fmt; if you're using go, use vim-go.

Reply:
```go
package main

func main() {

	defer func() {
		println("defer 1")
	}()

	defer func() {
		if r := recover(); r != nil {
			println("defer 2")
			panic("panic defer")
		}
	}()

	panic("panic")

}

// Output:
// defer 2
// defer 1

```

Reply:
done